programs/disks/DisksClient.h

```diff
@@ -74,15 +83,26 @@ class DisksClient
 
     void switchToDisk(const String & disk_, const std::optional<String> & path_);
 
+    std::vector<String> getInitializedDiskNames() const;
+    std::vector<String> getUninitializedDiskNames() const;
     std::vector<String> getAllDiskNames() const;
 
-    std::vector<String> getAllFilesByPatternFromAllDisks(const String & pattern) const;
+    // If ignore_exception is true, then the function will not throw an exception but can return incomplete result (though all returned paths are valid). This is useful for autocomplete which should not fail if there are any problems with disks.
+    std::vector<String> getAllFilesByPatternFromInitializedDisks(const String & pattern, bool ignore_exception) const;
 
+    void addDisk(String disk_name, std::optional<String> path);
 
-private:
-    void addDisk(DiskPtr disk_, const std::optional<String> & path_);
+    bool isDiskInitialized(const String & disk_name) const { return created_disks.contains(disk_name); }
 
+private:
     String current_disk;
-    std::unordered_map<String, DiskWithPath> disks;
+    std::unordered_map<String, DiskWithPath> disks_with_paths;
+    DisksMap created_disks;
```

Comment:
```suggestion
    DisksMap initialized_disks;
```
just a suggestion - it could be good to rename to match getInitializedDiskNames() and getUnitinializedDiskNames()